src/main/java/org/folio/service/mapping/RuleFactory.java

```diff
@@ -172,8 +180,44 @@ private boolean isTransformationValid(Transformations mappingTransformation) {
       LOGGER.error("Failed to fetch default rules for export");
       throw new NotFoundException(e);
     }
-    defaultRules = Arrays.asList(Json.decodeValue(stringRules, Rule[].class));
+    defaultRules = new ArrayList<>();
+    CollectionUtils.addAll(defaultRules, Json.decodeValue(stringRules, Rule[].class));
     return defaultRules;
   }
 
+  protected List<Rule> getDefaultHoldingsRulesFromFile() {
+    if (nonNull(holdingsDefaultRules)) {
+      return holdingsDefaultRules;
+    }
+    URL url = Resources.getResource(DEFAULT_HOLDINGS_RULES_PATH);
+    String stringRules = null;
+    try {
+      stringRules = Resources.toString(url, StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      LOGGER.error("Failed to fetch default holdings rules for export");
+      throw new NotFoundException(e);
+    }
+    holdingsDefaultRules = new ArrayList<>();
+    CollectionUtils.addAll(holdingsDefaultRules, Json.decodeValue(stringRules, Rule[].class));
+    return holdingsDefaultRules;
+  }
+
+  private List<Rule> createRulesDependsOnRecordType(MappingProfile mappingProfile) {
+    List<Rule> rules = new ArrayList<>();
+    if (mappingProfile == null || mappingProfile.getRecordTypes().contains(INSTANCE) || isEmptyTransformationsByRecordTypePresent(mappingProfile, INSTANCE)) {
```

Comment:
I think in case if empty transformations for Instance present we should generate rules using CombinedBuilder

Reply:
thanks, updated